docs/development/Architecture.md

```diff
@@ -0,0 +1,272 @@
+# This folder is not actually intended to be a regular package    #
+# HOWEVER, we need to keep this __init.py__ file in order to      #
+# make it visible by other modules.                               #
+# In future versions, this package should be moved to a lib so    #
+# that it will be an independent module instead a "false" package #
+###################################################################
+
+# DON'T REMOVE: Module importer overwritten to prevent bidirectional dependencies
+from _sl_build.secure_importer import override_module_importer
+override_module_importer()
+```
+
+#### Production Code Package
+**The name of this package must match the main package**. This is the place where the actual business logic code of the module is located. 
+The structure inside it is completely free, and it depends on only in the needs of the logic implemented in the module.
+
+#### Test Code Package
+As has been already stated along this document, each module is a completely independent piece of software. Of course, 
+this also includes that it must also contain its own test code. Since each module has specific responsibilities, these are 
+the only ones that need to be tested in them both from a unitary and integration perspective. For example, every SLP must 
+have unitary tests for the classes implemented there, but also integration tests that verifies that the whole conversion to 
+OTM process works for every identified casuistic.
+
+Regarding the structure of the whole test code package, there is no a hard constraint about it, but, for readability reasons, 
+it is recommended to follow a common one for every module. The one chosen is split in the following packages:
+> **Note**: Please don’t be fanatic with the definition of unitary and integration. There are not always clear boundaries, 
+> but common sense is the best way to identify what needs to be mocked or not.
+
+* `unit`. For unitary tests.
+* `integration`. For integration tests.
+* `resources`. It is also recommended to place here a test_resource_paths.py to have a unique point of access to the actual testing resources.
+* `util`. In case some general utilities need to be developed for being used in the tests, this package is the right place for them.**
+
+### Main Tests Package
+> **Note:**  It is necessary to evaluate the convenience to have this package, but, at least for now, it is important to keep 
+> this tests to guarantee the quality of the code.
+
+At the root level of the Startleft repository (the same as the other modules) is located a general tests package. Unlike 
+the test packages located inside each module, this one is generic to all the Startleft project and contains integration tests 
+that involves logic belonging to all the modules. By definition, this package can only contain integration tests. The boundaries 
+between this kind of tests and the ones developed by QA is hard to define, but it is clear that they are necessary, at least for now. 
```

Comment:
As QA testing is an internal process, should this mention be removed?

Reply:
Agree. Changed in the review branch.